### PR TITLE
Surface_mesh_approximation: Deal with boundary edges

### PR DESCRIPTION
**Pull request by afabri**
## Summary of Changes

Add a named parameter to distinguish between the approximation error for boundary and non-boundary edges of the input mesh. 

## Release Management

* Affected package(s): Surface_mesh_approximation
* Feature/Small Feature (if any): 
* Link to compiled documentation (obligatory for small feature) [*wrong link name to be changed*](httpssss://wrong_URL_to_be_changed/Manual/Pkg)
* License and copyright ownership:  unchanged

### CODE REVIEW

File: Surface_mesh_approximation/include/CGAL/Variational_shape_approximation.h [hunk 1]
```diff
@@ -1879,67 +1890,71 @@ class Variational_shape_approximation {
...
-        chord_vec = scale_functor(chord_vec, FT(1.0) / chord_len);
-        for (Boundary_chord_iterator citr = chord_begin; citr != chord_end; ++citr) {
-          Vector_3 vec = vector_functor(pt_begin, m_vpoint_map[target(*citr, *m_ptm)]);
-          vec = cross_product_functor(chord_vec, vec);
-          const FT dist = CGAL::approximate_sqrt(vec.squared_length());
-          if (dist > dist_max) {
-            chord_max = citr;
-            dist_max = dist;
-          }
+        FT dist_max(0.0);
```
Comment (sloriot) [inline Surface_mesh_approximation/include/CGAL/Variational_shape_approximation.h:1893]: unwanted indentation changes I guess

**Comment by sloriot** [comment]
Successfully tested in CGAL-6.0-Ic-20

**Comment by sloriot** [comment]
New warnings in Surface_mesh_approximation tests in CGAL-6.0-Ic-25

**Comment by sloriot** [comment]
Successfully tested in CGAL-6.0-Ic-33

**Comment by lrineau** [comment]
@MaelRL Do you remember why you had set https://github.com/CGAL/cgal/labels/Not%20yet%20approved?

**Comment by afabri** [comment]
> @MaelRL Do you remember why you had set [ Not yet approved](https://github.com/CGAL/cgal/labels/Not%20yet%20approved) The feature or pull-request has not yet been approved. ?

Because we still have to decide if the API shall also allow to constrain other than boundary vertices.

File: Surface_mesh_approximation/include/CGAL/Variational_shape_approximation.h [hunk 2]
```diff
@@ -827,9 +840,11 @@ class Variational_shape_approximation {
     using parameters::choose_parameter;
 
     const FT subdivision_ratio = choose_parameter(get_parameter(np, internal_np::subdivision_ratio), FT(5.0));
+    const FT boundary_subdivision_ratio = choose_parameter(get_parameter(np, internal_np::boundary_subdivision_ratio), FT(5.0));
```
Comment (afabri) [inline Surface_mesh_approximation/include/CGAL/Variational_shape_approximation.h:843]: In order to be backward compatible we should set `boundary_subdivision_ratio`to the value of `subdivision_ratio` as default.

Comment (afabri) [reply Surface_mesh_approximation/include/CGAL/Variational_shape_approximation.h:843]: @sloriot.  Is this a good idea?  This somehow assumes an order on named parameters.

File: Surface_mesh_approximation/include/CGAL/Variational_shape_approximation.h [hunk 3]
```diff
@@ -1854,8 +1871,16 @@ class Variational_shape_approximation {
...
+
+    if(is_boundary && boundary_subdivision_ratio == 0){
+      for (Boundary_chord_iterator citr = chord_begin; *citr != he_last; ++citr) {
+        attach_anchor(*citr);
+      }
+    }
+
     // do not subdivide trivial non-circular chord
-    if ((anchor_first != anchor_last) && (chord_size < 4))
+    if ((anchor_first != anchor_last) && (chord_size < 2))
```
Comment (MaelRL) [inline Surface_mesh_approximation/include/CGAL/Variational_shape_approximation.h:1883]: @palliez We should be able to reduce this to `chord_size == 1`? It's quite a simplification, so it feels like we maybe missed something? We checked the doc / code history to see where the `4` was coming from, but there's nothing... Any idea why we would *not* want to subdivide a chord if it has fewer than 4 edges?

File: Surface_mesh_approximation/include/CGAL/Variational_shape_approximation.h (hunk 2 above)
Comment (sloriot) [reply Surface_mesh_approximation/include/CGAL/Variational_shape_approximation.h]: This is correct, the variable `subdivision_ratio` being initialized before the call to the next get param.

**Comment by sloriot** [comment]
https://github.com/CGAL/cgal/pull/7574

**Comment by lrineau** [comment]
> > @MaelRL Do you remember why you had set [ Not yet approved](https://github.com/CGAL/cgal/issues?q=is%3Aopen+label%3A%22Not+yet+approved%22+sort%3Aupdated-desc) The feature or pull-request has not yet been approved. ?
> 
> Because we still have to decide if the API shall also allow to constrain other than boundary vertices.

@MaelRL: this PR is marked https://github.com/CGAL/cgal/labels/Tested, but also https://github.com/CGAL/cgal/labels/Not%20yet%20approved. I will not merge it, for now.

File: Surface_mesh_approximation/include/CGAL/Variational_shape_approximation.h (hunk 3 above)
Comment (MaelRL) [reply Surface_mesh_approximation/include/CGAL/Variational_shape_approximation.h:1883]: Discussed IRL and there was no trace of why it was `4`. Let's leave the newer `2`, then.

**Pull request merged**